cmd/skupper/skupper_cluster_test.go

```diff
@@ -0,0 +1,1579 @@
+			expectedOutput:  "",
+			expectedError:   "Failed to create connection token: Edge configuration cannot accept connections",
+			realCluster:     true,
+		},
+	}
+
+	namespace = "cmd-connection-token-edge-cluster-test-" + strings.ToLower(utils.RandomId(4))
+	kubeContext = ""
+	kubeConfigPath = ""
+
+	if *clusterRun {
+		cli = NewClient(namespace, kubeContext, kubeConfigPath)
+	} else {
+		cli = newMockClient(namespace)
+	}
+
+	if c, ok := cli.(*client.VanClient); ok {
+		_, err := kube.NewNamespace(namespace, c.KubeClient)
+		assert.Check(t, err)
+		defer kube.DeleteNamespace(namespace, c.KubeClient)
+
+	}
+	skupperInit(t, []string{"--edge"}...)
+
+	for _, tc := range testcases {
+		if tc.realCluster && !*clusterRun {
+			continue
+		}
+
+		cmd := NewCmdConnectionToken(testClient)
+		silenceCobra(cmd)
+		testCommand(t, cmd, tc.doc, tc.expectedError, tc.expectedCapture, tc.expectedOutput, tc.args...)
+	}
+}
+
+func TestConnectionTokenWithInteriorCluster(t *testing.T) {
+	testcases := []struct {
+		doc             string
+		args            []string
+		expectedCapture string
+		expectedOutput  string
+		expectedError   string
+		realCluster     bool
+	}{
+		{
+			doc:             "connection-token-test1",
+			args:            []string{"--help"},
+			expectedCapture: "",
+			expectedOutput:  "Create a connection token.  The 'connect' command uses the token to establish a connection from a remote Skupper site.",
+			expectedError:   "",
+			realCluster:     false,
+		},
+		{
+			doc:             "connection-token-test2",
+			args:            []string{},
+			expectedCapture: "",
+			expectedOutput:  "",
+			expectedError:   "accepts 1 arg(s), received 0",
+			realCluster:     false,
+		},
+		{
+			doc:             "connection-token-test3",
+			args:            []string{"/tmp/foo.yaml"},
+			expectedCapture: "Connection token written to /tmp/foo.yaml",
+			expectedOutput:  "",
+			expectedError:   "",
+			realCluster:     true,
+		},
+	}
+
+	namespace = "cmd-connection-token-interior-cluster-test-" + strings.ToLower(utils.RandomId(4))
+	kubeContext = ""
+	kubeConfigPath = ""
+
+	if *clusterRun {
+		cli = NewClient(namespace, kubeContext, kubeConfigPath)
+	} else {
+		cli = newMockClient(namespace)
+	}
+
+	if c, ok := cli.(*client.VanClient); ok {
+		_, err := kube.NewNamespace(namespace, c.KubeClient)
+		assert.Check(t, err)
+		defer kube.DeleteNamespace(namespace, c.KubeClient)
+
+	}
+	skupperInit(t, []string{"--cluster-local"}...)
+
+	for _, tc := range testcases {
+		if tc.realCluster && !*clusterRun {
+			continue
+		}
+
+		cmd := NewCmdConnectionToken(testClient)
+		silenceCobra(cmd)
+		testCommand(t, cmd, tc.doc, tc.expectedError, tc.expectedCapture, tc.expectedOutput, tc.args...)
+	}
+}
+
+func TestConnectWithEdgeCluster(t *testing.T) {
+	testcases := []struct {
+		doc             string
+		args            []string
+		expectedCapture string
+		expectedOutput  string
+		expectedError   string
+		realCluster     bool
+	}{
```

Comment:
just a comment: you could define a testCase type just private for this pks_test and save this boilerplate on every test

Reply:
Good suggestion, will change